src/core_modules/capture-core/components/LockedSelector/LockedSelector.epics.js

```diff
@@ -113,3 +113,15 @@ export const validateSelectionsBasedOnUrlUpdateEpic = (action$: InputObservable,
 
             return validSelectionsFromUrl();
         }));
+
+export const fetchOrgUnitEpic = (action$: InputObservable) =>
+    action$.pipe(
+        ofType(lockedSelectorActionTypes.FETCH_ORG_UNIT),
+        switchMap(({ payload: { orgUnitId } }) =>
+            from(fetchOrgUnits(orgUnitId))
```

Comment:
Should have been a comment on a previous PR, but I suggest we try using the app runtime data engine when we're quyering the api (replacing getApi). The epic signature would look something like [this](https://github.com/dhis2/capture-app/blob/0e6d99f8b3112e6a21699e50be4b82127d20b50a/src/core_modules/capture-core/components/Pages/MainPage/TeiWorkingLists/epics/teiViewEpics/teiView.epics.js#L9 ). From there you can use `querySingleResource`. At some point we can remove getApi altogether (and the entire old d2).

Reply:
Okay didnt realise that we dont wanna be using that anymore. Thanks!